pkg/pool-aclamm/test/foundry/AclAmmPoolVirtualBalances.t.sol

```diff
@@ -0,0 +1,258 @@
+            assertGe(
+                poolVirtualBalancesAfter[1],
+                poolVirtualBalancesBefore[1],
+                "Virtual B balance after should be greater than before"
+            );
+        }
+    }
+
+    function testSwap_Fuzz(uint256 exactAmountIn) public {
+        exactAmountIn = bound(exactAmountIn, 1e18, 10_000e18);
+
+        uint256[] memory virtualBalances = _calculateVirtualBalances();
+        uint256 invariantBefore = _getCurrentInvariant();
+
+        vm.prank(alice);
+        router.swapSingleTokenExactIn(pool, dai, usdc, exactAmountIn, 1, UINT256_MAX, false, new bytes(0));
+
+        uint256 invariantAfter = _getCurrentInvariant();
+        assertEq(invariantBefore, invariantAfter, "Invariant should not change");
+
+        uint256[] memory curentVirtualBalances = AclAmmPool(pool).getLastVirtualBalances();
+        assertEq(curentVirtualBalances[0], virtualBalances[0], "Virtual A balances don't equal");
+        assertEq(curentVirtualBalances[1], virtualBalances[1], "Virtual B balances don't equal");
+    }
+
+    function testAddLiquidity_Fuzz(uint256 exactBptAmountOut) public {
+        exactBptAmountOut = bound(exactBptAmountOut, 1e18, 10_000e18);
+
+        // uint256[] memory virtualBalances = _calculateVirtualBalances();
+
+        uint256 invariantBefore = _getCurrentInvariant();
+
+        vm.prank(alice);
+        router.addLiquidityProportional(
+            pool,
+            [MAX_UINT128, MAX_UINT128].toMemoryArray(),
+            exactBptAmountOut,
+            false,
+            new bytes(0)
+        );
+
+        uint256 invariantAfter = _getCurrentInvariant();
+
+        assertGt(invariantAfter, invariantBefore, "Invariant should increase");
+
+        // uint256[] memory curentVirtualBalances = AclAmmPool(pool).getLastVirtualBalances();
+        // assertEq(curentVirtualBalances[0], virtualBalances[0], "Virtual A balances don't equal");
+        // assertEq(curentVirtualBalances[1], virtualBalances[1], "Virtual B balances don't equal");
+    }
+
+    function testRemoveLiquidity_Fuzz(uint256 exactBptAmountIn) public {
+        exactBptAmountIn = bound(exactBptAmountIn, 1e18, 10_000e18);
+
+        // uint256[] memory virtualBalances = _calculateVirtualBalances();
+
+        uint256 invariantBefore = _getCurrentInvariant();
+
+        vm.prank(lp);
+        router.removeLiquidityProportional(
+            pool,
+            exactBptAmountIn,
+            [uint256(1), 1].toMemoryArray(),
+            false,
+            new bytes(0)
+        );
+
+        uint256 invariantAfter = _getCurrentInvariant();
+        assertLt(invariantAfter, invariantBefore, "Invariant should decrease");
+
+        // uint256[] memory curentVirtualBalances = AclAmmPool(pool).getLastVirtualBalances();
+        // assertEq(curentVirtualBalances[0], virtualBalances[0], "Virtual A balances don't equal");
+        // assertEq(curentVirtualBalances[1], virtualBalances[1], "Virtual B balances don't equal");
```

Comment:
Are these comments on purpose? Can we remove them?
If not, I'd leave a TODO here with some explanation on future work